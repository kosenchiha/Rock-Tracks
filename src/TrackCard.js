import React from 'react'
import { Link } from 'react-router-dom'

const TrackCard = ({ track }) => {
  return (
    <div
      className="d-flex border rounded m-2"
      style={{ width: '325px', background: '#eee' }}
    >
      <img
        className="m-2 rounded"
        height={100}
        src={track.artworkUrl100}
        alt=""
      />
      <div className="ml-2 p-2 d-flex flex-column justify-content-center">
        <p className="mb-0">{track.trackName}</p>
        <p className="mb-0">by {track.artistName}</p>
        <p className="mb-0">$ {track.trackPrice}</p>
        <Link to={`track/${track.trackId}`}>
          <p>Details ...</p>
        </Link>
      </div>
    </div>
  )
}

export default TrackCard
