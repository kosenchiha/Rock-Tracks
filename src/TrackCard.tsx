import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { TrackI } from './redux/tracksActionTypes'
import { RootStore } from './redux/store'
import { addLike, removeLike } from './redux/likesActions'
import heart from './heart.svg'
import heartGrey from './heart-grey.svg'

interface TrackCardProps {
  track: TrackI
}

const TrackCard: React.FC<TrackCardProps> = ({
  track: { artworkUrl100, trackName, artistName, trackPrice, trackId }
}) => {
  const dispatch = useDispatch()
  const isLiked = useSelector((state: RootStore): boolean =>
    state.likesState.likes.includes(trackId)
  )
  const onLike = () => {
    isLiked ? dispatch(removeLike(trackId)) : dispatch(addLike(trackId))
  }
  return (
    <div className="d-flex border rounded m-2 track-card">
      <div className="d-flex flex-column justify-content-center">
        <img className="m-2 rounded" height={100} src={artworkUrl100} alt="" />
        <img
          alt=""
          src={isLiked ? heart : heartGrey}
          className="mx-auto"
          height={70}
          width={70}
          onClick={onLike}
        />
      </div>
      <div className="ml-2 p-2 d-flex flex-column">
        <p className="mb-0 font-weight-bold">{trackName}</p>
        <p className="mb-0 track-data">by {artistName}</p>
        <p className="mb-0 track-data">$ {trackPrice}</p>
        <Link to={`track/${trackId}`}>
          <p>Details ...</p>
        </Link>
      </div>
    </div>
  )
}

export default TrackCard
