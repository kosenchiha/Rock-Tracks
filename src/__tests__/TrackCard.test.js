import React from 'react'
import { render } from '@testing-library/react'
import TrackCard from '../TrackCard.js'
import { BrowserRouter as Router } from 'react-router-dom'

describe('TrackCard', () => {
  const track = {
    wrapperType: 'track',
    trackId: 1513350083,
    artistName: 'Rock',
    trackName: 'I Am Rock',
    artistViewUrl: 'https://music.apple.com/us/artist/rock/218563?uo=4',
    collectionViewUrl:
      'https://music.apple.com/us/album/i-am-rock/1513349632?i=1513350083&uo=4',
    trackViewUrl:
      'https://music.apple.com/us/album/i-am-rock/1513349632?i=1513350083&uo=4',
    previewUrl:
      'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview113/v4/33/bd/29/33bd2943-3b25-80b8-1473-c30747726caf/mzaf_18234752672058510959.plus.aac.p.m4a',
    artworkUrl30:
      'https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/f4/66/73/f4667376-f04b-855d-6947-5787063fea2d/source/30x30bb.jpg',
    artworkUrl60:
      'https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/f4/66/73/f4667376-f04b-855d-6947-5787063fea2d/source/60x60bb.jpg',
    artworkUrl100:
      'https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/f4/66/73/f4667376-f04b-855d-6947-5787063fea2d/source/100x100bb.jpg',
    trackPrice: 1.29,
    releaseDate: '2005-09-06T12:00:00Z',
    collectionExplicitness: 'notExplicit',
    trackExplicitness: 'notExplicit',
    trackTimeMillis: 230827
  }

  const renderComponent = () => {
    return render(
      <Router>
        <TrackCard track={track} />
      </Router>
    )
  }

  it('matches snapshot', () => {
    const wrapper = renderComponent()
    expect(wrapper).toMatchSnapshot()
  })

  it('should display track name', () => {
    const { getByText } = renderComponent()
    const trackName = getByText(/I Am Rock/i)
    expect(trackName).toBeInTheDocument()
  })

  it('should display price', () => {
    const { getByText } = renderComponent()
    const price = getByText(/1.29/i)
    expect(price).toBeInTheDocument()
  })
})
