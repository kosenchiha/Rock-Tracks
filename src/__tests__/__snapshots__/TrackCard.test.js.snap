// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackCard matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex border rounded m-2 track-card"
      >
        <img
          alt=""
          class="m-2 rounded"
          height="100"
          src="https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/f4/66/73/f4667376-f04b-855d-6947-5787063fea2d/source/100x100bb.jpg"
        />
        <div
          class="ml-2 p-2 d-flex flex-column"
        >
          <p
            class="mb-0 font-weight-bold"
          >
            I Am Rock
          </p>
          <p
            class="mb-0 track-data"
          >
            by 
            Rock
          </p>
          <p
            class="mb-0 track-data"
          >
            $ 
            1.29
          </p>
          <a
            href="/track/1513350083"
          >
            <p>
              Details ...
            </p>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex border rounded m-2 track-card"
    >
      <img
        alt=""
        class="m-2 rounded"
        height="100"
        src="https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/f4/66/73/f4667376-f04b-855d-6947-5787063fea2d/source/100x100bb.jpg"
      />
      <div
        class="ml-2 p-2 d-flex flex-column"
      >
        <p
          class="mb-0 font-weight-bold"
        >
          I Am Rock
        </p>
        <p
          class="mb-0 track-data"
        >
          by 
          Rock
        </p>
        <p
          class="mb-0 track-data"
        >
          $ 
          1.29
        </p>
        <a
          href="/track/1513350083"
        >
          <p>
            Details ...
          </p>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
